src/redis_string.cc

```diff
@@ -63,15 +101,22 @@ rocksdb::Status String::Append(const std::string &user_key, const std::string &v
 }
 
 std::vector<rocksdb::Status> String::MGet(const std::vector<Slice> &keys, std::vector<std::string> *values) {
-  std::string ns_key;
-  std::string value;
-  std::vector<rocksdb::Status> statuses;
-  for (size_t i = 0; i < keys.size(); i++) {
-    AppendNamespacePrefix(keys[i], &ns_key);
-    statuses.emplace_back(getValue(ns_key, &value));
-    values->emplace_back(value);
+  std::vector<std::string> ns_keys;
+  ns_keys.reserve(keys.size());
+  for (const auto &key : keys) {
+    std::string ns_key;
+    AppendNamespacePrefix(key, &ns_key);
+    ns_keys.emplace_back(ns_key);
   }
-  return statuses;
+  std::vector<Slice> slice_keys;
+  slice_keys.reserve(ns_keys.size());
+  // don't use range-based for loop here, coz the slice member
+  // would refer the address instead of copy the value, and use
+  // range-based for loop may cause all members refer to the same addr
+  for (size_t i = 0; i < ns_keys.size(); i++) {
+    slice_keys.emplace_back(ns_keys[i]);
```

Comment:
why we use `slice_keys`, `ns_keys` is not enough?

Reply:
the RocksDB needs the slice vector instead of the string vector, so we need to convert the ns_keys to the slice vector here.